app/jobs/v3/update_service_instance_job.rb

```diff
@@ -41,7 +41,7 @@ def perform
           updates = message.updates.tap do |u|
             u[:service_plan_guid] = service_plan.guid
             u[:dashboard_url] = broker_response[:dashboard_url] if broker_response.key?(:dashboard_url)
-            u[:maintenance_info] = message.maintenance_info if message.maintenance_info
+            u[:maintenance_info] = message.maintenance_info || service_plan.maintenance_info
```

Comment:
I think there is a risk that the maintenance info will be changed even when the user is changing just parameters?
Am I right in thinking that the service plan ill be set even when the service plan is not being updated?

Scenario I am thinking:
SI is created with MI version 1.0.0
Plan is updated to MI version 1.0.0 
User requests parameter changes only
As a side effect the MI is also updated - which should not happen

Reply:
Good catch @FelisiaM!!! Will change that. Thanks!